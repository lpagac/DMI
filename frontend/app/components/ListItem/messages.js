/*
 * ListItem Messages
 *
 * This contains all the text for the ListItem component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.ListItem';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the ListItem component!',
  },
});
