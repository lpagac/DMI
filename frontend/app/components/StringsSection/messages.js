/*
 * StringsSection Messages
 *
 * This contains all the text for the StringsSection component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.StringsSection';

export default defineMessages({
  error: {
    id: `${scope}.error`,
    defaultMessage: 'Uh-Oh, No strings found. Try again later.',
  },
});
