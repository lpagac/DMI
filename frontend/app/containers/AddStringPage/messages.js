/*
 * AddStringPage Messages
 *
 * This contains all the text for the AddStringPage container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.AddStringPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Add A String!',
  },
  invalidInput: {
    id: `${scope}.invalidInput`,
    defaultMessage: 'Try adding some text!',
  },
  submit: {
    id: `${scope}.submit`,
    defaultMessage: 'ADD',
  },
  error: {
    id: `${scope}.error`,
    defaultMessage:
      'Unable to add string. Please check your input and try again.',
  },
  success: {
    id: `${scope}.success`,
    defaultMessage: 'Nice string! Add some more!',
  },
});
