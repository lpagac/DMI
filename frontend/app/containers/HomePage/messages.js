/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.HomePage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Check out our latest strings!',
  },
  subHeader: {
    id: `${scope}.subHeader`,
    defaultMessage: 'Limited time only... Add your own string!',
  },
});
