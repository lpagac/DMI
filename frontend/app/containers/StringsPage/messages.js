/*
 * StringsPage Messages
 *
 * This contains all the text for the StringsPage container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.StringsPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Pretty sweet collection of strings right?',
  },
});
