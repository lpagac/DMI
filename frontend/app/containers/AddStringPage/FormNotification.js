import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import {
  makeSelectAddingStringSuccess,
  makeSelectAddingStringError,
  makeSelectAddingString,
} from 'containers/App/selectors';

import LoadingIndicator from 'components/LoadingIndicator';
import Notification from 'components/Notification';
import messages from './messages';

export function FormNotification({
  addingString,
  addingStringSuccess,
  addingStringError,
}) {
  if (addingString !== false) {
    return (
      <div data-testid="loading-indicator">
        <LoadingIndicator />
      </div>
    );
  }

  if (addingStringError !== false) {
    return (
      <Notification success={false}>
        <FormattedMessage {...messages.error} />
      </Notification>
    );
  }

  if (addingStringSuccess !== false) {
    return (
      <Notification success>
        <FormattedMessage {...messages.success} />
      </Notification>
    );
  }

  return null;
}

FormNotification.propTypes = {
  addingString: PropTypes.bool,
  addingStringError: PropTypes.any,
  addingStringSuccess: PropTypes.bool,
};

export const mapStateToProps = createStructuredSelector({
  addingString: makeSelectAddingString(),
  addingStringSuccess: makeSelectAddingStringSuccess(),
  addingStringError: makeSelectAddingStringError(),
});

const withConnect = connect(mapStateToProps);

export default compose(withConnect)(FormNotification);
