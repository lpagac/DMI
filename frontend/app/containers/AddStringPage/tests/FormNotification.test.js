/**
 * Test the FormNotification
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { FormNotification } from '../FormNotification';
import configureStore from '../../../configureStore';
import messages from '../messages';

describe('<FormNotification />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <FormNotification
            addingString={false}
            addingStringSuccess={false}
            addingStringError={false}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should show success message when addingStringSuccess is true', () => {
    const utils = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <FormNotification
            addingString={false}
            addingStringError={false}
            addingStringSuccess
          />
        </IntlProvider>
      </Provider>,
    );

    const successMessage = utils.getByText(messages.success.defaultMessage);
    expect(successMessage).not.toEqual(null);
  });

  it('should show error message when addingStringError is true', () => {
    const utils = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <FormNotification
            addingString={false}
            addingStringSuccess={false}
            addingStringError
          />
        </IntlProvider>
      </Provider>,
    );

    const successMessage = utils.queryByText(messages.success.defaultMessage);
    const errorMessage = utils.getByText(messages.error.defaultMessage);
    expect(successMessage).toEqual(null);
    expect(errorMessage).not.toEqual(null);
  });

  it('should show loading indicator when addingString is true', () => {
    const utils = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <FormNotification
            addingString
            addingStringSuccess={false}
            addingStringError={false}
          />
        </IntlProvider>
      </Provider>,
    );

    const successMessage = utils.queryByText(messages.success.defaultMessage);
    const errorMessage = utils.queryByText(messages.error.defaultMessage);
    const loadingIndicator = utils.queryByTestId('loading-indicator');
    expect(successMessage).toEqual(null);
    expect(errorMessage).toEqual(null);
    expect(loadingIndicator).not.toEqual(null);
  });
});
