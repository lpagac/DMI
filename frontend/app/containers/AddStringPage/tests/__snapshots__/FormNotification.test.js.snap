// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormNotification /> should render and match the snapshot 1`] = `null`;
